lib/active_merchant/billing/gateways/stripe.rb

```diff
@@ -555,6 +556,22 @@ def add_emv_metadata(post, creditcard)
         post[:metadata][:card_read_method] = creditcard.read_method if creditcard.respond_to?(:read_method)
       end
 
+      def add_shipping_address(post, payment, options = {})
+        return unless shipping = options[:shipping_address]
+
+        post[:shipping] = {}
+
+        post[:shipping][:address] = {}
+        post[:shipping][:name] = shipping[:name] || options[:billing_address].try(:[], :name) || payment.try(:name)
```

Comment:
These changes were originally merged in #4484 but were later reverted because an attempt to use `options[:billing_address][:name] was resulting in a NoMethodError when neither `shipping[:name]` nor `[:billing_address]` were present.

Line 565 here is the primary difference between the last PR and this one. It guards against the NoMethodError with `.try()` and provides another fallback, which is to use the `name` stored with the payment method itself. (The `payment.try(:name)` is an attempt to guard against the possibility of a payment method not having a `:name` attribute, such as in the case of a third party token.)

The purpose of including these fallback options is that Stripe will fail the transaction if the shipping fields don't include a name. Until now, attempts to send shipping address to Stripe have been quietly ignored, so if this change did not include the fallback option, it's possible that some merchants might start to see Stripe failure messages for transactions that had previously succeeded. (In fact, I feel pretty confident that the ActiveMerchant errors that necessitated the earlier revert would have instead manifested as Stripe error messages -- either way, the transaction would not have succeeded.)

Part of me wonders if instead of adding this logic around filling in a shipping name, if we should check for `options[:shipping_address] && options[:shipping_address][:name]` in the guard clause on line 560 instead. (Or perhaps some other alternative?)

Reply:
I modified this (along with the test suite expectations) to return from the `add_shipping_address` method if no shipping name was present